### PR TITLE
DAH-1948: Upgrade Dependencies and Convert Puppeteer to Cypress

### PR DESCRIPTION
**Pull request by chadbrokaw**
[DAH-1948](https://sfgovdt.jira.com/browse/DAH-1948)

[DAH-1948]: https://sfgovdt.jira.com/browse/DAH-1948?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "javascript.validate.enable": true,
   "jest.disabledWorkspaceFolders": ["spec/javascript/e2e"],
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (chadbrokaw) [inline .vscode/settings.json:5]: These are the same thing: https://code.visualstudio.com/updates/v1_85#_code-actions-on-save-and-auto

**Review by tallulahkay — APPROVED** [review]
This lgtm!! Hefty pr, I tested around on the app and everything seems to work :D

**Review by alulabeshue-sfgov — COMMENTED** [review]
LGTM as well. A few comments / questions:

1. There are some remaining usage of var in babel.config.js
2. In package.json there are still some puppeteer deps, do we still need them?
3. Why did we remove the test splitting that we had before in circleci?

**Comment by chadbrokaw** [comment]
> LGTM as well. A few comments / questions:
> 
> 1. There are some remaining usage of var in babel.config.js
> 2. In package.json there are still some puppeteer deps, do we still need them?
> 3. Why did we remove the test splitting that we had before in circleci?

Thanks for taking a look!

1. Changed those, thanks for catching!
2. Oh I think those snuck back in after reverting a commit. We don't need them I think
3. Thank you for reminding me! I created a ticket to address that: https://sfgovdt.jira.com/browse/DAH-1991. I couldn't get the splitting to work with the Cypress tests. I will address that in 1991.

**Review by alulabeshue-sfgov — APPROVED** [review]
LG, thanks Chad!

**Pull request merged**